packages/kbn-shared-ux-components/src/empty_state/empty_state_page.stories.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { action } from '@storybook/addon-actions';
+import React from 'react';
+import { servicesFactory, DataServiceFactoryConfig } from '@kbn/shared-ux-storybook';
+import { SharedUxServicesProvider } from '@kbn/shared-ux-services';
+import mdx from './empty_state_page.mdx';
+import { NoDataPageProps } from '../page_template';
+import { EmptyStatePage } from './empty_state_page';
+
+export default {
+  title: 'Empty State Page',
+  description: 'A component to display when there is no data available',
+  parameters: {
+    docs: {
+      page: mdx,
+    },
+  },
+};
+
+const noDataConfig = {
+  solution: 'Analytics',
+  pageTitle: 'Analytics',
+  logo: 'logoKibana',
+  action: {
+    elasticAgent: {
+      title: 'Add Integrations',
+    },
+  },
+  docsLink: 'http://www.docs.com',
+};
+
+type Params = Pick<NoDataPageProps, 'solution' | 'logo' | 'pageTitle'> & DataServiceFactoryConfig;
+
+export const PureComponent = (params: Params) => {
+  const { solution, logo, pageTitle, hasESData, hasUserDataView } = params;
+  const serviceParams = { hasESData, hasUserDataView, hasDataViews: false };
+  const services = servicesFactory(serviceParams);
+  return (
+    <SharedUxServicesProvider {...services}>
+      <EmptyStatePage
```

Comment:
Can we have fallback content for when `hasESData` and `hasUserDataView` is true? Right now this component will just render nothing, so I'm confused as to how consumers expect to specifically use this in conjunction with their primary content.
<img width="1406" alt="Screen Shot 2022-04-25 at 13 01 58 PM" src="https://user-images.githubusercontent.com/549577/165138068-0bb6cc21-d8e1-4341-b8a3-7955c78b7370.png">

Reply:
I am not sure how this would work yet. Plugins will need to perform some sort of a check when this should be displayed. I agree taking a child component as an optional prop to be rendered when neither option is true makes sense, but it might not be needed yet? I would leave it as-is now and address it later, once we know better what is needed here.

Reply:
As long as we do come back to address it.